dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -323,18 +351,42 @@ await foreach (Completions completions in response.ConfigureAwait(false))
         for (int requestIndex = 1; ; requestIndex++)
         {
             // Make the request.
-            var responseData = (await RunRequestAsync(() => this.Client.GetChatCompletionsAsync(chatOptions, cancellationToken)).ConfigureAwait(false)).Value;
-            this.CaptureUsageDetails(responseData.Usage);
-            if (responseData.Choices.Count == 0)
+            ChatCompletions? responseData = null;
+            List<OpenAIChatMessageContent> responseContent;
+            using (var activity = ModelDiagnostics.StartCompletionActivity(this.Endpoint, this.DeploymentOrModelName, ModelProvider, chat, executionSettings))
```

Comment:
As written, we're going to get a span around each roundtrip to the LLM, but if there are N roundtrips as part of the single user call to GetChatMessageContentsAsync, those N roundtrips won't be wrapped in a single activity. Can we add another layer of activity around the loop, such that all the activities for each constituent requests are grouped?

Reply:
I agree that we should wrap all the auto tool calls in an activity. I will do that in a separate PR. I created an item to track the work: https://github.com/microsoft/semantic-kernel/issues/6176